backend/src/SerialHandler.py

```diff
@@ -0,0 +1,265 @@
+
+        encBuf = Codec.Encode(buf, len(buf), ProtoCore.MessageID.MSG_COMMAND)
+        target_enum = utl.get_node_from_str(target)
+        if (target_enum == ProtoCore.NODE_DMB or target_enum == ProtoCore.Node.NODE_PBB) and self.port == RADIO_SERIAL_PORT:
+            self.serial_port.write(encBuf)
+        if (target_enum ==  ProtoCore.NODE_RCU or target_enum ==  ProtoCore.Node.NODE_SOB) and self.port == UART_SERIAL_PORT:
+            self.serial_port.write(encBuf)
+
+        return True
+
+# Procedures =======================================================================================
+
+def serial_rx_thread(ser_han: SerialHandler):
+    """
+    Thread function for the incoming serial data listening.
+    """
+    while (ser_han.kill_rx):
+        ser_han.handle_serial_message()
+        pass
+
+def process_serial_workq_message(message: WorkQ_Message, ser_han: SerialHandler) -> bool:
+        """
+        Process the message from the workq.
+
+        Args:
+            message (str):
+                The message from the workq.
+        """
+        logger.debug(f"Processing serial workq message: {message.message_type}")
+        messageID = message.message_type
+
+        if messageID == THREAD_MESSAGE_KILL:
+            logger.debug(f"Killing {ser_han.thread_name} thread")
+            return False
+        elif messageID == THREAD_MESSAGE_SERIAL_WRITE:
+            command = message.message[0]
+            target = message.message[1]
+            command_param = message.message[2]
+            source_sequence_number = message.message[3]
+            ser_han.send_serial_command_message(command, target, command_param, source_sequence_number)
+        return True
+
+def serial_thread(thread_name: str, device: SerialDevices, baudrate: int, thread_workq: mp.Queue, message_handler_workq: mp.Queue):
+    """
+    Thread function for the incoming serial data listening.
+
+    Args:
+        thread_name (str):
+            The name of the thread.
+        device (SerialDevices):
+            The device to listen to.
+        baudrate (int):
+            The baudrate of the serial connection.
+        thread_workq (mp.Queue):
+            The workq to send messages to the thread.
+        message_handler_workq (mp.Queue):
+            The workq to send messages to the database handler. 
+    """
+    if device == SerialDevices.UART:
+        port = UART_SERIAL_PORT
+    elif device == SerialDevices.RADIO:
+        port = RADIO_SERIAL_PORT
+    
+    serial_workq = thread_workq
+    ser_han = SerialHandler(thread_name, port, baudrate, message_handler_workq)
+    if ser_han.serial_port == None:
+        return
+    
+    rx_thread = threading.Thread(target=serial_rx_thread, args=(ser_han,))
+    rx_thread.start()
+    while (1):
+        # If there is any workq messages, process them first
+        # then once the queue is empty read the serial port
+        if not serial_workq.empty():
```

Comment:
Since we don't need to poll a separate thing now let's get rid of this empty and block forever for a workq item, this'll also give the rx thread more time to run.